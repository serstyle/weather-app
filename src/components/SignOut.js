import React from 'react'
import { Redirect } from 'react-router-dom'

class SignOut extends React.Component{
	state = {id: '',
			email: '',
			name: '',
			city: 'Paris'
	}


	componentDidMount(){
		this.props.getUser(this.state)
		
	}


	render(){
		return(<Redirect to='/weather-app' />)
	}
}

export default SignOut;