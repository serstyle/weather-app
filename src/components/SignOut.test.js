import React from 'react'
import {shallow} from 'enzyme'
import SignOut from './SignOut'

const mockGetUser = jest.fn()
const wrap = shallow(<SignOut getUser={mockGetUser} />)

it('snapshot componenet', ()=>{
    expect(wrap).toMatchSnapshot()
    expect(mockGetUser).toHaveBeenCalledTimes(1);
})