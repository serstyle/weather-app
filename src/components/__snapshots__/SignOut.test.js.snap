// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot componenet 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SignOut
    getUser={
      [MockFunction] {
        "calls": Array [
          Array [
            Object {
              "city": "Paris",
              "email": "",
              "id": "",
              "name": "",
            },
          ],
        ],
        "results": Array [
          Object {
            "isThrow": false,
            "value": undefined,
          },
        ],
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "push": false,
      "to": "/",
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "push": false,
        "to": "/",
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
